src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,225 @@
+        printLineSpacer();
+        Scanner scanner = new Scanner(System.in);
+        String userLineInput;
+        int numOfTasks = 0;
+
+        numOfTasks = getFile(userLists, numOfTasks);
+
+        do {
+            userLineInput = scanner.nextLine();
+            String[] splitInputs = userLineInput.split(" ");
+            String userCommand = splitInputs[0];
+            printLineSpacer();
+            if (!userCommand.equals(("bye"))) {
+                try {
+                    switch (userCommand) {
+                    case "list":
+                        printList(userLists, numOfTasks);
+                        break;
+                    case "done":
+                        try {
+                            completeTask(userLists, splitInputs[1]);
+                        } catch (ArrayIndexOutOfBoundsException | NullPointerException e) {
+                            System.out.println("Please choose a valid list entry.");
+                        }
+                        break;
+                    case "event":
+                    case "deadline":
+                    case "todo":
+                        try {
+                            storeTasks(userLists, userLineInput, numOfTasks, userCommand);
+                            numOfTasks++;
+                        } catch (StringIndexOutOfBoundsException e ) {
+                            System.out.println(TEXT_SPACER +
+                                    "Description of your task can't be empty. At least enter something!");
+                        }
+                        break;
+                    case "delete":
+                        try {
+                            deleteEvent(userLists, splitInputs, numOfTasks);
+                            numOfTasks--;
+                        } catch (ArrayIndexOutOfBoundsException | NullPointerException e){
+                            System.out.println(TEXT_SPACER + "Please choose a valid entry to delete.");
+                        }
+                        break;
+                    default:
+                        throw new DukeExceptions(TEXT_SPACER +
+                                "Sorry mate, I don't understand what you are trying to get me to do.");
+                    }
+
+                    saveToFile(userLists, numOfTasks);
+                } catch (DukeExceptions | IOException e) {
+                    System.out.println(e.getMessage());
+                }
+                printLineSpacer();
+
+            }
+        } while(!userLineInput.equals("bye"));
+
+        System.out.println(TEXT_SPACER + "Aight. See you soon mate.");
+        printLineSpacer();
+    }
+
+    private static int getFile(ArrayList userLists, int numOfTasks) {
+        try {
+            File saveFile = new File(path + "/task.txt");
+            saveFile.createNewFile();
+            Scanner savedTasks = new Scanner(saveFile);
+            while(savedTasks.hasNext()) {
```

Comment:
For consistency, include a character space before the `while` condition. E.g. `while (savedTasks.hasNext()) {`.